File: Array/Reverse.cpp
#include<iostream>
using namespace std;

class Array
{
    public:
        int iSize;
        int *Arr;

        Array(int iNo)//constructor
        {
            iSize = iNo;
            Arr= new int[iSize];
        }

        ~Array()//destructor
        {
            delete []Arr;   
        }

        void Accept()
        {
            cout<<"Enter the elements : "<<"\n";

            int iCnt = 0;
            for(iCnt=0;iCnt<iSize;iCnt++)
            {
                cin>>Arr[iCnt];
            }
        }

        void Display()
        {
            cout<<"Elements of the array are : "<<"\n";

            int iCnt = 0;
            for(iCnt=0;iCnt<iSize;iCnt++)
            {
                cout<<Arr[iCnt]<<"\n";
            }
        }

        void Reverse()
        {
            int start = 0;
            int end = iSize - 1;

            while (start < end) {
                // Swap elements at start and end indices
                int temp = Arr[start];
                Arr[start] = Arr[end];
                Arr[end] = temp;

                // Move towards the center of the array
                start++;
                end--;
            }
            cout<<"Elements of the array after reversing : ";
        }
};

int main()
{
    int iNo=0;
    int iRet = 0,iRet1=0;

    cout<<"Enter number of elements : "<<"\n";
    cin>>iNo;

    Array aobj(iNo);
    aobj.Accept();
    aobj.Display();
    aobj.Reverse();
    aobj.Display();

    return 0;
}

File: Array/Reverse.java
import java.util.*;

class ArrayX
{
    public void Reverse(int Arr[])
    {
        int start = 0;
        int end = iSize - 1;

        while (start < end) {
            // Swap elements at start and end indices
            int temp = Arr[start];
            Arr[start] = Arr[end];
            Arr[end] = temp;

            // Move towards the center of the array
            start++;
            end--;
        }
        cout<<"Elements of the array after reversing : ";
    }
}
class Reverse
{
    public static void main(String Arg[])
    {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the size of array: ");
        int iSize = sc.nextInt();

        int Arr[] = new int[iSize];

        System.out.println("Enter elements of the array : ");
        for(int iCnt = 0;iCnt<iSize;iCnt++)
        {
            Arr[iCnt]=sc.nextInt();
        }

        System.out.println("Enter elements of the array : ");
        for(int iCnt = 0;iCnt<iSize;iCnt++)
        {
            System.out.print(Arr[iCnt]+" ");
        }
        System.out.println();

        ArrayX obj = new ArrayX();
        obj.Reverse(Arr);

        System.out.println("Enter elements of the array : ");
        for(int iCnt = 0;iCnt<iSize;iCnt++)
        {
            Arr[iCnt]=sc.nextInt();
        }
    }
}